plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3673,6 +3673,20 @@ public void testEmptyDelete()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testInsertingIntoSpanshotIsNotPossible()
+    {
+        String tableName = "test_insert_into_snapshot_" + randomTableSuffix();
+        assertUpdate(format("CREATE TABLE %s (col int)", tableName));
+        assertUpdate(format("INSERT INTO %s VALUES 1,2,3", tableName), 3);
+        long snapshotId = getCurrentSnapshotId(tableName);
+        assertUpdate(format("INSERT INTO %s VALUES 4,5,6", tableName), 3);
+        assertQuery(format("SELECT * FROM \"%s@%d\"", tableName, snapshotId), "VALUES 1,2,3");
+        assertThatThrownBy(() -> query(format("INSERT INTO \"%s@%d\" VALUES 7,8,9", tableName, snapshotId)))
+                .hasMessage("Inserting data into snapshot is not supported in Iceberg.");
```

Comment:
Don't we need to reject other write operations?

Reply:
Yes, thank you for pointing this. My vision was tunnelled as always.